### DISCUSSION TITLE
Some QOL things

### DISCUSSION THREAD

**Discussion by phooenixdassmatte** [DFHack/dfhack#4308] [Ideas]
Okey this turned out to be more than i initially thought, only suggestions and an open dialoge ofc. I think i've gone through the whole DF hack menu's to find solutions to these issues but i might have missed some, so i apologize beforehand if that may be so. I also want to say that playing without DF hack is like having a second monitor for your computer, once you have gotten used to it you never wanna go back, an elogue to the team making this mod. 🥇 

**Furniture placement QOL:**
1. Have the option to show in menu (when selecting what specific table,chair,bed to place there) to sort by value instead of distance, i know where my furniture stockpile is so the distance is going to be the same for everything either way and for me atleast is pretty irrelevant. Yes you can see quality of say a steel table which is an indicator of value, but when you have encrusts and gems and stuff on it, makes it more complicated. I'm familiar with the regular quality indicators but makes it harder when it shows *<<≡Item Name≡>>*  and ≡<<*Item Name*>>≡ , and makes it even harder to read if you have 40 different encrusted steel tables in scrambled order and you're trying to figure out the one with the highest value.

I looked around in the planner if i would find anything there that solves my problem, but it doesn't really make me see the value of what i'm placing as it's just sorting encrusted or not, and material.

2. Bonus is if you could also inspect the furniture from the place menu aswell and go back (without closing the furniture selection window!) if you for example want to only place a specific furniture with a specific gem because that person who lives there loves a specific gem.

3. Being able to mass-remove furniture so you don't have to manually click on every one and designate to remove it, similar when you remove walls, stairs etc. Yes you can make a dump close and then mass dump them but kind of looked more for a feature than a work-around.

**Trade QOL:**
1. You can select stockpiles for trade which is very neat. Would be nice to also be able to set stockpiles to not even show up in the menu (the DF hack menu "Move goods to/from depot"). 
I have different stockpiles for finished goods but i only want one of them to show up, but i don't want to set it to autohaul as there are too many boxes to haul back and fourth every time and not even sell 1/10 of them, and often there is enough worn clothes to sell so i won't even need them. 

2. A new filter in the same menu to toggle smeltable yes or no. So when selling worn clothes you won't sell stuff from your smeltstockpile (that is if the above setting isn't implementet, as that smelt stockpile could be set to "don't trade" so they wouldn't show up) But could still be useful in other cases.

3. Some option to see the name of the stockpile the item/bin is in, and being able to sort items alphabetically by stockpile name, so everything from each stockpile will be "bundled together" in the list when sorting that way. As i tend to name all my stockpiles and have a kind of a "mind map" of what everything is, would make it easy to get a bigger picture of what everything is. As there are multiple different stockpiles with bins named "finished goods" for example. Now that i think about it this kinds of deals with both of the problems mentioned above, or atleast makes them alot easier to deal with.

**Comment by myk002** [comment]
> sort by value

This is a good idea, though adding this information to the vanilla screen may prove difficult. It would be easier to integrate into the `buildingplan` item selection screens (where we could also add distance info). I added https://github.com/DFHack/dfhack/issues/4312 for this

> Bonus is if you could also inspect the furniture from the place menu

This, unfortunately, is probably not going to happen, at least not until DF has a better way to open up item detail screens. Right now there is no good programmatic way to bring up unit/item/building info sheets. We can zoom to the location easily, but not actually bring up the info sheet panel.

> Being able to mass-remove furniture

Ctrl-m on the main map is the hotkey to bring up `gui/mass-remove`

> Trade QOL

I agree that (3) would probably be the most versatile option (and probably also easiest to implement). I filed https://github.com/DFHack/dfhack/issues/4313 for this

**Reply by phooenixdassmatte** [reply]
> Ctrl-m on the main map is the hotkey to bring up gui/mass-remove

Haha there is tools everywhere, thanks.
